rskj-core/src/main/java/co/rsk/net/sync/BlockConnectorHelper.java

```diff
@@ -0,0 +1,79 @@
+package co.rsk.net.sync;
+
+import co.rsk.core.BlockDifficulty;
+import org.ethereum.core.Block;
+import org.ethereum.db.BlockStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+
+public class BlockConnectorHelper {
+    private static final Logger logger = LoggerFactory.getLogger("SnapBlockConnector");
+    private final BlockStore blockStore;
+    private final List<BlockAndDifficulty> blockAndDifficultiesList;
+    public BlockConnectorHelper(BlockStore blockStore, List<BlockAndDifficulty> blockAndDifficultiesList) {
+        this.blockStore = blockStore;
+        this.blockAndDifficultiesList = blockAndDifficultiesList;
+        blockAndDifficultiesList.sort(new BlockAndDiffComparator());
+    }
+
+    public void startConnecting() {
+        Block child = null;
+        logger.info("Start connecting Blocks");
+        if (blockAndDifficultiesList.isEmpty()) {
+            return;
+        }
+        int blockIndex = blockAndDifficultiesList.size() - 1;
+        if (blockStore.isEmpty()) {
+            BlockAndDifficulty blockAndDifficulty = blockAndDifficultiesList.get(blockIndex);
+            child = blockAndDifficulty.getBlock();
+            logger.debug("BlockStore is empty, setting child block number the last block from the list: {}",child.getNumber());
+            blockStore.saveBlock(child, blockAndDifficulty.getDifficulty(), true);
+            blockIndex--;
+        } else {
+            logger.debug("BlockStore is not empty, getting best block");
+            child = blockStore.getBestBlock();
+            logger.debug("Best block number: {}", child.getNumber());
+        }
+        while (blockIndex >= 0) {
+            BlockAndDifficulty currentBlockAndDifficulty = blockAndDifficultiesList.get(blockIndex);
+            Block currentBlock = currentBlockAndDifficulty.getBlock();
+            logger.info("Connecting block number: {}", currentBlock.getNumber());
+
+            if (!currentBlock.isParentOf(child)) {
+                logger.error("Block is not parent of child");
```

Comment:
Here I would suggest the same as [here](https://github.com/rsksmart/rskj/pull/2301/files#r1591156205)